pkg/storage/bloom/v1/bloom_tester.go

```diff
@@ -0,0 +1,156 @@
+package v1
+
+import (
+	"github.com/prometheus/prometheus/model/labels"
+
+	"github.com/grafana/loki/pkg/logql/syntax"
+	"github.com/grafana/loki/pkg/storage/bloom/v1/filter"
+)
+
+type BloomTest interface {
+	Test(bloom filter.Checker) bool
+	TestWithPrefixBuf(bloom filter.Checker, buf []byte, prefixLen int) bool
+}
+
+type BloomTests []BloomTest
+
+func (b BloomTests) Test(bloom filter.Checker) bool {
+	for _, test := range b {
+		if !test.Test(bloom) {
+			return false
+		}
+	}
+	return true
+}
+
+func (b BloomTests) TestWithPrefixBuf(bloom filter.Checker, buf []byte, prefixLen int) bool {
+	for _, test := range b {
+		if !test.TestWithPrefixBuf(bloom, buf, prefixLen) {
+			return false
+		}
+	}
+	return true
+}
+
+func FiltersToBloomTest(b NGramBuilder, filters ...syntax.LineFilterExpr) BloomTest {
+	tests := make(BloomTests, 0, len(filters))
+	for _, f := range filters {
+		if f.Left != nil {
+			tests = append(tests, FiltersToBloomTest(b, *f.Left))
+		}
+		if f.Or != nil {
+			left := FiltersToBloomTest(b, *f.Or)
+			right := simpleFilterToBloomTest(b, f.LineFilter)
+			tests = append(tests, newOrTest(left, right))
+			continue
+		}
+
+		tests = append(tests, simpleFilterToBloomTest(b, f.LineFilter))
+	}
+	return tests
+}
+
+func simpleFilterToBloomTest(b NGramBuilder, filter syntax.LineFilter) BloomTest {
+	switch filter.Ty {
+	case labels.MatchEqual, labels.MatchNotEqual:
+		var test BloomTest = newStringTest(b, filter.Match)
+		if filter.Ty == labels.MatchNotEqual {
+			test = newNotTest(test)
+		}
+		return test
+	case labels.MatchRegexp, labels.MatchNotRegexp:
+		// TODO(salvacorts): Simplify regex similarly to how it's done at pkg/logql/log/filter.go (`simplify` function)
+		// 					 Ideally we want to extract the simplify logic into pkg/util/regex.go
+		return NoopTest
+	default:
+		return NoopTest
+	}
+}
+
+type noopTest struct{}
```

Comment:
nit like above: this could be named `matchAll`

Reply:
Done